internal/node/advancer/advancer.go

```diff
@@ -0,0 +1,108 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package advancer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/cartesi/rollups-node/internal/node/advancer/service"
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/cartesi/rollups-node/internal/node/nodemachine"
+)
+
+var (
+	ErrInvalidMachines   = errors.New("machines must not be nil")
+	ErrInvalidRepository = errors.New("repository must not be nil")
+	ErrInvalidAddress    = errors.New("no machine for address")
+)
+
+type Advancer struct {
+	machines   Machines
+	repository Repository
+}
+
+func New(machines Machines, repository Repository) (*Advancer, error) {
```

Comment:
There will be a "Manager" before advancers being created of killed, checking if is there a new Applications, right?